src/rp2_common/hardware_i2c/i2c.c

```diff
@@ -75,6 +75,13 @@ uint i2c_set_baudrate(i2c_inst_t *i2c, uint baudrate) {
     invalid_params_if(I2C, hcnt < 8);
     invalid_params_if(I2C, lcnt < 8);
 
+    // The SDA hold time should be at least 300 ns, per the I2C standard.
+    // sda_hold_count [cycles] = freq_in [cycles/s] * 300ns * (1s / 1e9ns)
+    // Reduce 300/1e9 to 3/1e7 to avoid numbers that don't fit in uint.
+    // Add 1 to avoid division truncation.
+    uint sda_hold_count = ((freq_in * 3) / 10000000) + 1;
```

Comment:
Perhaps this should be explicitly named `sda_tx_hold_count`, to match up with `I2C_IC_SDA_HOLD_IC_SDA_TX_HOLD_BITS` ?

Reply:
Good idea. `sda_hold_count` has been renamed to `sda_tx_hold_count`.